velox/type/FloatingPointUtil.h

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <array>
+#include <cmath>
+#include <vector>
+
+namespace facebook::velox {
+
+/// Custom comparator and hash functors for floating point types. These are
+/// designed to ensure consistent NaN handling according to the following rules:
+/// - NaN == NaN returns true, even for NaNs with differing binary
```

Comment:
Another question, if -NaN == NaN , then is sorting on these values deterministic or indeterministic (if you dont knwo the order of the Nan's).

Reply:
In IEEE-754 there is no notion of a negative NaN, the sign bit exists and some programs print it out, but for all intents and purposes a NaN is just a Not-a-Number. There exists a classification of NaNs (signalling and quiet) but those specify range of values, but the sign is of no consequence.

To clarify, the `-` in this comment is a dash, not a negative sign. If that caused any confusion, I can use something like `->` instead

Reply:
I was under impression it was a minus 😄